message.go

```diff
@@ -31,16 +36,68 @@ func (msg Message) item() messageSetItem {
 }
 
 func (msg Message) message() message {
+	var attrs int8
+	if msg.CompressionCodec != nil {
+		attrs = int8(msg.CompressionCodec.Code()) & compressionCodecMask
+	}
+
 	m := message{
-		MagicByte: 1,
-		Key:       msg.Key,
-		Value:     msg.Value,
-		Timestamp: timestamp(msg.Time),
+		MagicByte:  1,
+		Key:        msg.Key,
+		Value:      msg.Value,
+		Timestamp:  timestamp(msg.Time),
+		Attributes: attrs,
 	}
 	m.CRC = m.crc32()
 	return m
 }
 
+func (msg Message) encode() (Message, error) {
+	//if msg.CompressionCodec.Code() == CompressionNoneCode {
+	if msg.CompressionCodec == nil {
+		return msg, nil
+	}
+
+	codec, ok := codecs[msg.CompressionCodec.Code()]
```

Comment:
Is this necessary? The message already carries a codec, we could use it directly instead of forcing a check on the registered codecs.